compiler/rustc_hir_analysis/src/check/compare_impl_item.rs

```diff
@@ -183,13 +181,13 @@ fn compare_method_predicate_entailment<'tcx>(
             kind: impl_m.kind,
         });
 
-    // Create mapping from impl to placeholder.
-    let impl_to_placeholder_args = GenericArgs::identity_for_item(tcx, impl_m.def_id);
-
     // Create mapping from trait to placeholder.
-    let trait_to_placeholder_args =
-        impl_to_placeholder_args.rebase_onto(tcx, impl_m.container_id(tcx), trait_to_impl_args);
-    debug!("compare_impl_method: trait_to_placeholder_args={:?}", trait_to_placeholder_args);
+    let trait_to_impl_args = GenericArgs::identity_for_item(tcx, impl_m.def_id).rebase_onto(
```

Comment:
(preexisting) wouldn't `impl_m_args` or `impl_item_args` be a better fit over `trait_to_impl_args`? the trait args aren't actually involved

Reply:
I think that would be confusing because after this renaming, we'd be instantiating `impl_m_predicates` with identity args, and `trait_m_predicates` with `impl_m_args`...

Reply:
Fair enough, makes sense